Sources/NeedleFoundation/Internal/DependencyProviderRegistry.swift

```diff
@@ -68,6 +81,19 @@ public class __DependencyProviderRegistry {
         }
     }
 
+    /// Retrieve the dependency provider for the given componentpath.
+    ///
+    /// - parameter componentpath: The component path that uses the returned dependency provider.
+    /// - returns: The dependency provider for the given componentpath.
+    public func dependencyProviderFactory(`for` componentPath: String) -> ((Scope) -> AnyObject)? {
+        providerFactoryLock.lock()
+        defer {
+            providerFactoryLock.unlock()
+        }
+        
+        return providerFactories[componentPath.hashValue]
+    }
+
```

Comment:
Let's group the `public` methods together above the internal ones. And let's add a `// MARK: - Internal`.

Reply:
Updating.